androidApp/src/main/java/dev/johnoreilly/confetti/speakers/SpeakerAndroidViewModel.kt

```diff
@@ -0,0 +1,45 @@
+package dev.johnoreilly.confetti.speakers
+
+import com.rickclephas.kmm.viewmodel.coroutineScope
+import com.rickclephas.kmm.viewmodel.stateIn
+import dev.johnoreilly.confetti.ApolloClientCache
+import dev.johnoreilly.confetti.SpeakersViewModel
+import dev.johnoreilly.confetti.account.AccountUiState
+import dev.johnoreilly.confetti.account.Authentication
+import dev.johnoreilly.confetti.wear.WearSettingsSync
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+class SpeakerAndroidViewModel @Inject constructor(
+    val authentication: Authentication,
+    val apolloClientCache: ApolloClientCache,
+    val wearSettingsSync: WearSettingsSync
+): SpeakersViewModel() {
+
+    val accountUiState: StateFlow<AccountUiState> =
+        combine(authentication.currentUserFlow, wearSettingsSync.wearNodes) { user, wearNodes ->
+            AccountUiState(user, wearNodes)
+        }.stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), AccountUiState())
+
+    fun signOut() {
+        apolloClientCache.clear()
+        authentication.signOut()
+    }
+
+    fun installOnWear() {
+        viewModelScope.coroutineScope.launch {
+            accountUiState.value.wearNodes.filter { !it.isAppInstalled }.forEach {
+                wearSettingsSync.installOnWearNode(it.id)
+            }
+        }
+    }
+
+    fun updateWearTheme() {
```

Comment:
I think this duplication is a step in the wrong direction.  I wonder if we can split this into Scaffold and Individual screens.  But as is, I wouldn't add this.  It feels like we have got the lines drawn in slightly the wrong place currently, and it's visible in your PR.

How should screens have common functionality for android?

Reply:
> I wonder if we can split this into Scaffold and Individual screens

Some screens ("login", "conferences") do not have a `Scaffold` because they use the whole screen. But even if we had, the 1:1 mapping between screen and viewmodel somewhat enforces duplication of `FooViewModel.updateWearTheme()` and others...

Maybe for account/wear we use singleton (maybe even POJO?) state holders 

```kotlin
    val viewModel: SessionAndroidViewModel = getViewModel<SessionAndroidViewModel>()
    val uiState by viewModel.uiState.collectAsStateWithLifecycle()

    // account and wear we can reuse in every screen that has an account button
    val accountHolder: AccountHolder = koinInject()
    val accountUiState by accountHolder.state.collectAsStateWithLifecycle()

    val wearHolder: WearHolder = koinInject()
    val wearUiState by wearHolder.state.collectAsStateWithLifecycle()

```

Reply:
I can try it out with the stateHolder, that would at least reduce the duplication. If I'm splitting up the Scaffold with it's own VM I think I'm kinda back to AccountViewModel.

Reply:
I'm still unclear whether Account/Wear must be ViewModels in the `KMMViewModel`. The advantage is that we get the nice `@UiState` for iOS out of the box but it feels like we're abusing a bit the ViewModel cause these are things that are not specifically bound to a View.

Reply:
Very nice PR! Really cool to see how you solved it! :)

Reply:
Well TBH I'm not sure. This solves the bookmark sync issue but it makes everything more verbose because we now need to pass `uid` (and tokenProvider) all the way down. Let's wait for @yschimke input and if he's ok, let's do something similar for `Wear`? Maybe even using `WearSettingsSync` and `koinInject` directly from the `ConfettiScafflod` ?

Reply:
Yeah let's do that :) Is there any guidelines from koin to use the `koinInject`  directly in Compose like this? I could not find anything on their pages. Do you know? :)

Reply:
I think I found some here: https://insert-koin.io/docs/reference/koin-compose/compose although they where not that describing, I think you are correct :)

Reply:
Sorry, missed this.

Reply:
I'm very open to the koinInject. There is an open debate in the compose world on the value of ViewModels, especially when Compose is used cross platform.  Maybe time to explore this ourselves.

Reply:
I think part of the question for Confetti is how to bridge that to SwiftUI as we're reluying on  https://github.com/rickclephas/KMM-ViewModel to bridge to observable SwiftUI models. But since Wear is most likely Android only we don't need KMMViewModels there.